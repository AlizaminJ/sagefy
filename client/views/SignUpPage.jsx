import React from 'react'
import { shape, string } from 'prop-types'
import Layout from './components/Layout'
import Icon from './components/Icon'
import FormErrorsTop from './components/FormErrorsTop'
import FormErrorsField from './components/FormErrorsField'
import ExternalLink from './components/ExternalLink'

export default function SignUpPage({
  hash,
  gqlErrors,
  body: { name, email },
  query: { redirect },
}) {
  return (
    <Layout
      hash={hash}
      page="SignUpPage"
      title="Join Sagefy"
      description="Get access to the best learning content with your Sagefy account. Continue where you left off on your learning journey."
    >
      <FormErrorsTop formErrors={gqlErrors} />
      <FormErrorsField formErrors={gqlErrors} field="all" />

      <section>
        <h1>
          Join Sagefy <Icon i="signUp" s="h1" />
        </h1>

        <p>
          Already have an account?{' '}
          <a href={redirect ? `/log-in?redirect=${redirect}` : '/log-in'}>
            <Icon i="logIn" /> Log In
          </a>
          .<br />
          By signing up, you agree to our{' '}
          <a href="/terms">
            <Icon i="terms" /> Terms of Service
          </a>
          .
        </p>

        <form action="" method="POST">
          <p>
            <label htmlFor="name">Name</label>
            <input
              id="name"
              name="name"
              placeholder="example: Unicorn"
              value={name}
              type="text"
              size="40"
              autoFocus
              required
            />
          </p>
          <FormErrorsField formErrors={gqlErrors} field="name" />
          <p>
            <label htmlFor="email">Email</label>
            <input
              id="email"
              name="email"
              placeholder="example: unicorn@example.com"
              value={email}
              type="email"
              size="40"
              required
            />
            <br />
            <small>
              We need your email to send notices and to reset your password.
            </small>
          </p>
          <FormErrorsField formErrors={gqlErrors} field="email" />
          <p>
            <label htmlFor="password">Password</label>
            <input
              id="password"
              name="password"
              type="password"
              size="40"
              required
              pattern=".{8,}"
            />
            <br />
            <small>
              Please{' '}
              <ExternalLink href="https://heiskr.com/useapasswordmanager/">
                use a password manager
              </ExternalLink>
              .
            </small>
          </p>
          <FormErrorsField formErrors={gqlErrors} field="password" />
          <p>
            <button type="submit">
              <Icon i="signUp" /> Sign Up
            </button>
          </p>
        </form>
      </section>
    </Layout>
  )
}

SignUpPage.propTypes = {
  hash: string.isRequired,
  gqlErrors: shape({}),
  body: shape({}),
  query: shape({}).isRequired,
}

SignUpPage.defaultProps = {
  gqlErrors: {},
  body: {},
}
