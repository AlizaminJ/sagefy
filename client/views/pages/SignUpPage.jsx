import React from 'react'
import { Link } from 'react-router-dom'
import Icon from '../components/Icon'

export default function SignUpPage() {
  // TODO if I'm already logged in, redirect instead.
  return (
    <div className="SignUpPage">
      <section>
        <h1>
          Join Sagefy <Icon i="signUp" s="xxl" />
        </h1>

        <p>
          Already have an account?{' '}
          <Link to="/log-in">
            <Icon i="logIn" /> Log In
          </Link>
          .<br />
          By signing up, you agree to our
          <Link to="/terms">
            <Icon i="terms" /> Terms of Service
          </Link>
          .
        </p>

        <form action="" method="POST">
          <p>
            <label htmlFor="name">Name</label>
            <input
              id="name"
              name="name"
              placeholder="example: Unicorn"
              type="text"
              size="40"
              autoFocus
              required
            />
          </p>
          <p>
            <label htmlFor="email">Email</label>
            <input
              id="email"
              name="email"
              placeholder="example: unicorn@example.com"
              type="email"
              size="40"
              required
            />
            <br />
            <small>
              We need your email to send notices and to reset your password.
            </small>
          </p>
          <p>
            <label htmlFor="password">Password</label>
            <input
              id="password"
              name="password"
              type="password"
              size="40"
              required
              pattern=".{8,}"
            />
          </p>
          <p>
            <button type="submit">
              <Icon i="signUp" /> Sign Up
            </button>
          </p>
        </form>
      </section>
    </div>
  )
}
