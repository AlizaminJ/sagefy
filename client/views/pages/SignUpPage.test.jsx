import SignUpPage from './SignUpPage'

describe('SignUpPage', () => {
  it('should render the sign up page', () => {
    expect(
      SignUpPage({ gqlErrors: {}, prevValues: {}, query: {} })
    ).toMatchSnapshot()
  })
})
