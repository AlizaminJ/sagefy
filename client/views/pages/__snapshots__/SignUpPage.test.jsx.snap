// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage should render the sign up page 1`] = `
<Layout hash={hash}
  description="Get access to the best learning content with your Sagefy account. Continue where you left off on your learning journey."
  page="SignUpPage"
  title="Join Sagefy"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Join Sagefy
      <Icon
        i="signUp"
        s="xxl"
      />
    </h1>
    <p>
      Already have an account?

      <Link
        replace={false}
        to="/log-in"
      >
        <Icon
          i="logIn"
          s="m"
        />
         Log In
      </Link>
      .
      <br />
      By signing up, you agree to our

      <Link
        replace={false}
        to="/terms"
      >
        <Icon
          i="terms"
          s="m"
        />
         Terms of Service
      </Link>
      .
    </p>
    <form
      action=""
      method="POST"
    >
      <p>
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          autoFocus={true}
          id="name"
          name="name"
          placeholder="example: Unicorn"
          required={true}
          size="40"
          type="text"
        />
      </p>
      <FormErrorsField
        field="name"
        formErrors={Object {}}
      />
      <p>
        <label
          htmlFor="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          placeholder="example: unicorn@example.com"
          required={true}
          size="40"
          type="email"
        />
        <br />
        <small>
          We need your email to send notices and to reset your password.
        </small>
      </p>
      <FormErrorsField
        field="email"
        formErrors={Object {}}
      />
      <p>
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          pattern=".{8,}"
          required={true}
          size="40"
          type="password"
        />
      </p>
      <FormErrorsField
        field="password"
        formErrors={Object {}}
      />
      <p>
        <button
          type="submit"
        >
          <Icon
            i="signUp"
            s="m"
          />
           Sign Up
        </button>
      </p>
    </form>
  </section>
</Layout>
`;
